airflow-core/src/airflow/api_fastapi/core_api/security.py

```diff
@@ -322,6 +322,18 @@ def inner(
     return inner
 
 
+def requires_authenticated() -> Callable:
```

Comment:
I don't _love_ this, but it does work. If there is a better/cleaner/easier way, I'm all ears.

Reply:
Via middleware?

Reply:
Middleware could be a good option
Middleware is applying to all endpoints, right? In this case, we may want to limit if the endpoint is /config. Additionally, if we don't have an early exit, when it is not /config endpoint, It could also increase the response time a bit, right? I know it is async, but async not to block the application, but the request will still wait for its response

Reply:
We will stick with this, but good thought.